tests/audio/test_pit.py

```diff
@@ -0,0 +1,167 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import namedtuple
+from functools import partial
+from typing import Callable
+
+import numpy as np
+import pytest
+import torch
+from scipy.optimize import linear_sum_assignment
+from torch import Tensor
+
+from tests.helpers import seed_all
+from tests.helpers.testers import BATCH_SIZE, NUM_BATCHES, MetricTester
+from torchmetrics.audio import PIT
+from torchmetrics.functional import pit, si_sdr, snr
+from torchmetrics.utilities.imports import _TORCH_GREATER_EQUAL_1_6
+
+seed_all(42)
+
+Time = 10
+
+Input = namedtuple('Input', ["preds", "target"])
+
+inputs1 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, Time),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, Time),
+)
+inputs2 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, Time),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, Time),
+)
+
+
+def scipy_version(preds: Tensor, target: Tensor, metric_func: Callable, eval_func: str):
+    batch_size, spk_num = target.shape[0:2]
+    metric_mtx = torch.empty((batch_size, spk_num, spk_num), device=target.device)
+    for t in range(spk_num):
+        for e in range(spk_num):
+            metric_mtx[:, t, e] = metric_func(preds[:, e, ...], target[:, t, ...])
+
+    # pit_r = PIT(metric_func, eval_func)(preds, target)
+    metric_mtx = metric_mtx.detach().cpu().numpy()
+    best_metrics = []
+    best_perms = []
+    for b in range(batch_size):
+        row_idx, col_idx = linear_sum_assignment(metric_mtx[b, ...], eval_func == 'max')
+        best_metrics.append(metric_mtx[b, row_idx, col_idx].mean())
+        best_perms.append(col_idx)
+    return torch.from_numpy(np.stack(best_metrics)), torch.from_numpy(np.stack(best_perms))
+
+
+def average_metric(preds: Tensor, target: Tensor, metric_func: Callable):
+    # shape: preds [BATCH_SIZE, 1, Time] , target [BATCH_SIZE, 1, Time]
+    # or shape: preds [NUM_BATCHES*BATCH_SIZE, 1, Time] , target [NUM_BATCHES*BATCH_SIZE, 1, Time]
+    return metric_func(preds, target)[0].mean()
+
+
+snr_pit_scipy = partial(scipy_version, metric_func=snr, eval_func='max')
+si_sdr_pit_scipy = partial(scipy_version, metric_func=si_sdr, eval_func='max')
```

Comment:
```suggestion
scipy_snr_pit = partial(scipy_version, metric_func=snr, eval_func='max')
scipy_si_sdr_pit = partial(scipy_version, metric_func=si_sdr, eval_func='max')
```
lest be consistent with other `sklearn`s

Reply:
the name `scipy_version ` changed to `a_naive_implementation_of_pit_based_on_scipy` in https://github.com/PyTorchLightning/metrics/pull/384/commits/29744ab59371fdef6fd1e3b7b803b2a72b6078a8. maybe here we can usse `snr_naive_pit` and `si_sdr_naive_pit`?